### PR TITLE
refactor: removes unnecessary array operations

### PR DESCRIPTION
**Pull request by celsobonutti**
As pointed out by @fakenickels in [#181](https://github.com/Astrocoders/reform/issues/181)

### CODE REVIEW

File: packages/bs-reform/__tests__/SchemaDemo_test.re [hunk 1]
```diff
@@ -30,7 +30,9 @@ test("should validate a correct record", () => {
   module ProfileValidation = ReSchema.Make(ProfileLenses);
 
   let schema =
-    ProfileValidation.Validation.Schema([|Email(Email), IntMin(Age, 18)|]);
+    ProfileValidation.Validation.(
+      Schema(Belt.Array.concat(email(Email), int(~min=18, Age)))
```
Comment (fakenickels) [inline packages/bs-reform/__tests__/SchemaDemo_test.re:34]: the local open of `Validation.(` already allows you to use the custom infix I provided just for that
```suggestion
      Schema(email(Email) + int(~min=18, Age))
```

File: packages/bs-reform/src/ReForm.re [hunk 2]
```diff
@@ -453,14 +449,10 @@ module Make = (Config: Config) => {
 
       send(SetFieldsState(newFieldsState));
 
-      Belt.Array.keep(newFieldsState, ((field, _)) =>
+      Belt.Array.keepMap(newFieldsState, ((field, fieldState)) => {
```
Comment (fakenickels) [inline packages/bs-reform/src/ReForm.re:452]: amazing 🙏

File: packages/bs-reform/src/ReForm.re [hunk 3]
```diff
@@ -418,14 +418,10 @@ module Make = (Config: Config) => {
...
                 let newFieldState =
-                  fieldsValidated
-                  ->Belt.Array.keep(fieldStateValidated =>
-                      Belt.Option.map(fieldStateValidated, ((item, _)) =>
-                        item
-                      )
-                      == Some(field)
-                    )
-                  ->Belt.Array.get(0);
+                  fieldsValidated->Belt.Array.getBy(fieldStateValidated =>
```
Comment (fakenickels) [inline packages/bs-reform/src/ReForm.re:421]: sweet

File: packages/reschema/src/ReSchema.re [hunk 4]
```diff
@@ -314,9 +310,9 @@ module Make = (Lenses: Lenses) => {
       );
 
     let errors =
-      validationList
-      ->Belt.Array.keep(((_field, fieldState)) => fieldState !== Valid)
-      ->Belt.Array.map(((field, fieldState)) => (field, fieldState));
+      validationList->Belt.Array.keep(((_field, fieldState)) =>
```
Comment (fakenickels) [inline packages/reschema/src/ReSchema.re:313]: 👀

**Comment by fakenickels** [comment]
@all-contributors add @celsobonutti for code

File: packages/reschema/src/ReSchema.re (hunk 4 above)
Comment (celsobonutti) [reply packages/reschema/src/ReSchema.re:313]: 👀

**Pull request merged**